dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1359,131 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
+    def _get_correlation(self, clean_samples):
+        """
+        Calculate correlation matrix on the cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        columns = self.options.correlation.columns
+        if columns is None:
+            for col in self._profile:
+                if (self._profile[col].profile['data_type'] not in ['int', 'float']
+                        or self._profile[col].null_count > 0):
+                    clean_samples.pop(col)
+        if len(clean_samples) <= 1:
+            return
+
+        data = pd.DataFrame(clean_samples)
+        data = data.apply(pd.to_numeric, errors='coerce')
+        self.correlation_matrix = data.corr()
+
+    def _update_correlation(self, clean_samples):
+        """
+        Update correlation matrix for cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        if self.total_samples > 0:
+            # currently return None with the exising correlation
+            # TODO: implement the update with the exising correlation
+            warnings.warn("Currently, the updating correlations is disabled "
+                          "and will remain unchanged from its current value. "
+                          "Updating correlations will be available in a "
+                          "future update.")
+            return
```

Comment:
I believe this is still dangerous to set the value to the previous when we update as it gives a false expectation of what the correlations could be.

i would prefer to set the value to None if it gets updated.

Reply:
This also means the warning message needs to change.

Reply:
fixed